### PR TITLE
Implementing random request IDs

### LINKED ISSUE #96: Make error responses more informational
The following functions have been identified as having the potential to trigger an error.  When these errors happen, they should provide more information to the API user instead of just saying that the it was a bad request or internal server error.  The user will not have enough information at their fingertips and won't be able to divine the way to correct the request.

```
challenge-manager.js - validateAsync()
validation-manager.js - secureAsync() - twice
validate-request-middleware
```

### PR DESCRIPTION
**Pull request by fidian**
* Request IDs feed into error responses as the logref.  This removes the previous implementation of randomly assigned logref values.
* Fixed promise generation during tests so Node 6.6 and later do not emit warnings about unhandled and asynchronously handled Promise objects.
* Fixed problems where errors were generated and no useful information was returned from the API.
* Removed CORS support since the plugin was broken.

This closes #96.

### CODE REVIEW

**Comment by fidian** [comment]
Log time to DOT-313

**Comment by fidian** [comment]
Updated to address code review concerns.  Also added a test to increase the branch-related test coverage.

File: lib/middleware/request-logger-middleware.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        var d;
+
+        d = new Date();
+        logger.info(`${this.logId} ${d.toISOString()} ${message}`);
+    }
+
+
+    idLength = config.server.requestIdLength;
+
+    /**
```
Comment (quantumew) [inline lib/middleware/request-logger-middleware.js]: I think another newline above the JSDoc and one less newline above `idLength = config.server.requestIdLength;`

File: lib/storage/s3.js [hunk 2]
```diff
@@ -9,22 +9,20 @@
...
  *
- * @typedef {Object} s3~configOptions
- * @property {string} bucket
- * @property {string} region
+ * @typedef {Object} storage~putOptions
+ * @property {string} [contentType]
+ * @property {OtDate} [expires]
  */
 
 /**
```
Comment (quantumew) [inline lib/storage/s3.js]: Two newlines above these JSDoc blocks.

**Comment by fidian** [comment]
Fixed everything except the spaces after a function.  The style guide says there should be 2 blank lines after functions.  Weirdly, that note is under "Comments".  https://tests-always-included.github.io/style-guide/general/#comments

**Comment by fidian** [comment]
Merged!

**Pull request merged**